monai/networks/layers/convutils.py

```diff
@@ -93,11 +97,74 @@ def gaussian_1d(sigma: Union[float, torch.Tensor], truncated: float = 4.0) -> np
         1D torch tensor
 
     """
-    sigma = torch.as_tensor(sigma).float()
-    if sigma <= 0 or truncated <= 0:
+    sigma = torch.as_tensor(sigma, dtype=torch.float)
+    if sigma <= 0.0 or truncated <= 0.0:
         raise ValueError(f"sigma and truncated must be positive, got {sigma} and {truncated}.")
     tail = int(sigma * truncated + 0.5)
-    x = torch.arange(-tail, tail + 1).float()
-    t = 1 / (torch.tensor(2.0).sqrt() * sigma)
-    out = 0.5 * ((t * (x + 0.5)).erf() - (t * (x - 0.5)).erf())
-    return out.clamp(min=0)
+    if approx.lower() == "simple":
+        x = torch.arange(-tail, tail + 1, dtype=torch.float)
+        t = 1.0 / (torch.tensor(2.0).sqrt() * sigma)
+        out = 0.5 * ((t * (x + 0.5)).erf() - (t * (x - 0.5)).erf())
+        return out.clamp(min=0)
+    if approx.lower() == "refined":
+        out = [_modified_bessel_0(sigma), _modified_bessel_1(sigma)]
```

Comment:
Looks like there is a confusion between standard deviation and variance. Note that ITK asks for variance in the Gaussian operator: https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/Core/Common/include/itkGaussianOperator.hxx